Fix Issue #651

weissreto opened the pull request:
This changeset fixes issue #651. 

The old implementations only set the meta data if the value is not 0. So a value of 0 was not propagated to and considered by the client side validation. Moreover the value returned by getMinimum() is depending on the implementation of javax.faces.validator.LongRangeValidator. Some implmenations return a 0. Others returns Long.MIN_VALUE. 

The validators now set boolean flags if a minimum or maximum value are set. This flags are considered when building the meta data. The meta data is only set if also the minimum or maximum were set.

weissreto committed:
Fix Issue #651

tandraschko commented:
I would use Integer instead of int, so you don't need the additional flag. WDYT @cagataycivici?

weissreto committed:
Improve patch as suggested by tandraschko:
- Instead of boolean flags use Long, Double, Integer variable to either store the value or be null if no value was set.

cagataycivici commented:
Thanks for the feedback.
